### PR TITLE
Flask lesson

### PR DESCRIPTION
**Pull request by nataliaruzhytska**

### CODE REVIEW

File: flask_lesson_intro/main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def get_home_page():
+    return render_template("home.html", data=get_data())
+
+
+@app.route('/Alarm_clock')
```
Comment (dimigor) [inline flask_lesson_intro/main.py]: please follow convention, name URL with snake_case

File: flask_lesson_intro/main.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@app.route('/Battery_charger')
+def get_battery_charger():
+    return render_template('battery_charger.html', data=get_data())
+
+
+@app.route('/author_page')
+def get_author_page():
+    return render_template('/author_page.html')
```
Comment (dimigor) [inline flask_lesson_intro/main.py:28]: you do not need all this routes with templates. Make universal route

File: flask_lesson_intro/templates/base.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  <!-- Collapsible content -->
+  <div class="collapse navbar-collapse" id="basicExampleNav">
+
+    <!-- Links -->
+    <ul class="navbar-nav mr-auto">
+      <li class="nav-item">
+        <a class="nav-link" href="/Alarm_clock">Alarm clock</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="/Headphones">Headphones</a>
```
Comment (dimigor) [inline flask_lesson_intro/templates/base.html]: use `use url_for` instead

File: flask_lesson_intro/utils.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import json
+
+
+def get_data():
+    with open("data.json") as file:
+        return json.load(file)
```
Comment (dimigor) [inline flask_lesson_intro/utils.py]: what if the file will not found or file contains not valid data?

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_lesson_intro/main.py (hunk 2 above)
Comment (nataliaruzhytska) [reply flask_lesson_intro/main.py:28]: done

Comment (nataliaruzhytska) [reply flask_lesson_intro/main.py:28]: done

File: flask_lesson_intro/templates/item.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{% extends "base.html" %}
+
+{% block content %}
+
+    {% for i in data %}
+        {% if i['title'] == item %}
```
Comment (dimigor) [inline flask_lesson_intro/templates/item.html]: maybe this logic you should do at the route?

File: flask_lesson_intro/utils.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import json
+import os
+
+
+def get_data(data_path="data.json"):
+    if os.path.isfile(data_path):
```
Comment (dimigor) [inline flask_lesson_intro/utils.py]: you may just catch `(ValueError, IOError)`

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_lesson_intro/templates/item.html (hunk 5 above)
Comment (nataliaruzhytska) [reply flask_lesson_intro/templates/item.html]: please explain what do you mean

Comment (nataliaruzhytska) [reply flask_lesson_intro/templates/item.html]: done

**Review by dimigor — APPROVED** [review]

**Pull request merged**